Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -77,65 +77,107 @@ public actor KafkaProducer {
     /// Dictionary containing all topic names with their respective `rd_kafka_topic_t` pointer.
     private var topicHandles: [String: OpaquePointer]
 
-    // We use implicitly unwrapped optionals here as these properties need to access self upon initialization
-    /// Used for handling the connection to the Kafka cluster.
-    private var client: KafkaClient!
-    /// Task that polls the Kafka cluster for updates periodically.
-    private var pollTask: Task<Void, Never>!
+    /// Variable that indicates if our poll loop should be running.
+    private var polling = true
```

Comment:
We don't need this property. We can use the state for this `started` indicates that we are polling